src/main/java/com/nirima/jenkins/plugins/docker/DockerCloud.java

```diff
@@ -214,11 +214,14 @@ public DockerTemplate getTemplate(String template) {
     }
 
     /**
-     * Gets {@link DockerTemplate} that has the matching {@link Label}.
+     * Gets {@link DockerTemplate} that has the matching {@link Label}; if {@link Label} is null and togglemode false,
+     * then get this template.
      */
     public DockerTemplate getTemplate(Label label) {
         for (DockerTemplate t : templates) {
-            if(label == null || label.matches(t.getLabelSet())) {
+            if(label == null && !t.getToggleMode()) {
+                return t;
+            } else if (label.matches(t.getLabelSet())) {
```

Comment:
The elseif will give a Null pointer exception if the slave is set to exclusive

+1 to combining
```
if( (label == null && !t.getExclusiveMode()) || (label != null && label.matches(t.getLabelSet())) ) {
```